python/src/robyn/modeling/pareto/response_curve.py

```diff
@@ -239,7 +239,7 @@ def _check_metric_dates(
             date_range_updated = all_dates[metric_loc]
         elif date_range.startswith("last_"):
             n_periods = int(date_range.split("_")[1])
-            metric_loc = slice(end_rw - n_periods + 1, end_rw)
+            metric_loc = slice(end_rw - n_periods, end_rw + 1)
```

Comment:
will this break the pareto notebook?

Reply:
Actually also my suggestion was wrong @alxlyj .

As the `slice(start,stop)` method in python, `start` is inclusive and `stop` is not as described [here.](https://docs.python.org/3/library/stdtypes.html#sequence-types) I'd keep the same pattern also for `rolling_window_start_which` and `rolling_window_end_which`, which implies this code:

`metric_loc = slice(end_rw - n_periods, end_rw)`

I also fix this line:
`date_range.startswith("last_"):` by adding a check first: `elif isinstance(date_range, str) and ` as if one passed the date_range as a list, the code thrown an exception